sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,500 @@
+outstanding `async` invocations that have not been balanced by a matching `get`
+call. In the presence of this property, potentially expensive hardware
+implementing the destructor may be elided. It is undefined behavior to specify
+the `balanced` property on `task_sequence` and then to allow the `task_sequence`
+object to be destroyed while there are any `async` invocations for which `get`
+has not been called.
+
+== `task_sequence` Properties
+
+The following code and table describe the "balanced", "invocation_capacity", and
+"response_capacity" properties that can be provided when declaring a
+`task_sequence`.
+
+[source,c++,linenums]
+----
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace experimental {
+
+struct balanced_key {
+  using value_t = property_value<balanced_key>;
+};
+
+struct invocation_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<invocation_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+struct response_capacity_key {
+  template <uint32_t Size>
+  using value_t = property_value<response_capacity_key,
+    std::integral_constant<uint32_t, Size>>;
+};
+
+inline constexpr balanced_key::value_t balanced;
+
+template <> struct is_property_key<balanced_key> : std::true_type {};
+
+// TODO: Not sure if the template here should just be
+// template<auto &f, class propertiesT>
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<balanced_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+template <uint32_t Size>
+inline constexpr invocation_capacity_key::value_t<Size> invocation_capacity;
+
+template <> struct is_property_key<invocation_capacity_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<invocation_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+template <uint32_t Size>
+inline constexpr response_capacity_key::value_t<Size> response_capacity;
+
+template <> struct is_property_key<response_capacity_key> : std::true_type {};
+
+template <typename ReturnT, typename ... ArgsT, ReturnT(&f) (ArgsT...),
+  class propertiesT>
+struct is_property_key_of<response_capacity_key,
+  task_sequence<f, propertiesT>> : std::true_type {};
+
+} // namespace experimental
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+----
+
+--
+[options="header"]
+|===
+| Property | Description
+| balanced | The `balanced` property is a guarantee to the SYCL device compiler
+that the `task_sequence` object will call exactly the same number of `async` s
+and `get` s over the object's lifetime (i.e. before the `task_sequence`
+destructor is invoked). 
+
+| invocation_capacity | The `invocation_capacity` property defines the minimum
+required number `Size` of `async` invocations that must be invokable without any
```

Comment:
This wording is confusing.  I think an earlier version may have stated this more clearly.  Something along these lines would be more readable:

The 'invocation_capacity' property defines the minimum number of 'async'd invocations that may be outstanding at any time without causing async calls to block.  An outstanding invocation is a call to async whose corresponding result has not yet been retrieved by a call to get().

Reply:
Updated to say so.